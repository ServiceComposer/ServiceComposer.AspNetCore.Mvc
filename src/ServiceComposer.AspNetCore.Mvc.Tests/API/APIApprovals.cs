using ApprovalTests;
using ApprovalTests.Reporters;
using PublicApiGenerator;
using System.Runtime.CompilerServices;
using Xunit;

namespace ServiceComposer.AspNetCore.Mvc.Tests.API
{
    public class APIApprovals
    {
        [Fact]
        [UseReporter(typeof(DiffReporter))]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void Approve_API()
        {
            var publicApi = typeof(IHandleResult).Assembly.GeneratePublicApi();

            Approvals.Verify(publicApi);
        }
    }
}
