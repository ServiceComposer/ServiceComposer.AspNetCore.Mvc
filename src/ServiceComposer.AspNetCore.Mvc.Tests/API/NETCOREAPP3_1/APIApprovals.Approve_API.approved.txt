[assembly: System.Runtime.CompilerServices.InternalsVisibleTo("ServiceComposer.AspNetCore.Mvc.Tests")]
[assembly: System.Runtime.Versioning.TargetFramework(".NETCoreApp,Version=v3.1", FrameworkDisplayName="")]
namespace ServiceComposer.AspNetCore.Mvc
{
    public interface IHandleResult : ServiceComposer.AspNetCore.IInterceptRoutes
    {
        System.Threading.Tasks.Task Handle(string requestId, Microsoft.AspNetCore.Mvc.Filters.ResultExecutingContext context, [System.Runtime.CompilerServices.Dynamic] object viewModel, int httpStatusCode);
    }
    public class ViewModelCompositionMvcOptions
    {
        public void RegisterResultHandler(System.Type type) { }
        public void RegisterResultHandler<T>()
            where T : ServiceComposer.AspNetCore.Mvc.IHandleResult { }
    }
    public static class ViewModelCompositionOptionsExtensions
    {
        public static void AddMvcSupport(this ServiceComposer.AspNetCore.ViewModelCompositionOptions compositionOptions) { }
        public static void AddMvcSupport(this ServiceComposer.AspNetCore.ViewModelCompositionOptions compositionOptions, System.Action<ServiceComposer.AspNetCore.Mvc.ViewModelCompositionMvcOptions> config) { }
    }
}